assets/js/src/common/set_from_address_modal.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import ReactDOMServer from 'react-dom/server';
+import ReactStringReplace from 'react-string-replace';
+import MailPoet from 'mailpoet';
+
+const showSetFromAddressModal = async () => {
```

Comment:
Can I ask why did you mark this function as `async`? There is nothing wrong with it, but it doesn't return anything (if I didn't miss anything) so I don't see the reason for it.